tensorflow/lite/micro/kernels/split_v_test.cc

```diff
@@ -0,0 +1,606 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/c/builtin_op_data.h"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/micro/all_ops_resolver.h"
+#include "tensorflow/lite/micro/debug_log.h"
+#include "tensorflow/lite/micro/kernels/kernel_runner.h"
+#include "tensorflow/lite/micro/testing/micro_test.h"
+#include "tensorflow/lite/micro/testing/test_utils.h"
+
+namespace tflite {
+namespace testing {
+
+void TestSplitVThreeOutputsFloat(
```

Comment:
It would be nice if these functions (`TestSplitVThreeOutputsFloat`. `TestSplitVTwoOutputsFloat`, `TestSplitVEightOutputsFloat`) can be combined into one function template.

Maybe the param to the function template is a struct template with the template parameter being the number of outputs?

Something like:

```cc
template<int N>
struct OutputTensors {
float *data[N];
int *dims[N] 
}

template<int N>
void TestSplitVFloat(const OutputTensors<N>& output_tensors, ...)

```

I haven't thought through all the details so maybe there is something that prevents this. Happy to go deeper on this if it helps -- more readable and maintainable testing code something we would like to move towards.

Reply:
Done.
Not sure if this is more readable but I definitely agree this will be much more maintainable with less code duplication.
I added the expect output data arrays to the struct to make the function calls a bit more readable. See if this is ok or if you'd rather I did something else.
Thanks :)

Reply:
Thanks for all the changes, looks good to me.